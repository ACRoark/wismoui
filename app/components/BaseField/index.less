@import (reference) '../../styles/themes/index.less';

@base-field-cls-prefix: ~'@{dte-prefix}-base-field';

.@{base-field-cls-prefix} {
  &-children {
    color: @dte-gray-600;
  }

  &-label {
    color: @dte-gray-700;
    font-size: @dte-wismo-font-size-1;
    font-weight: @dte-wismo-font-weight-semi-bold;
    padding-bottom: 6px;
  }

  &-section {
    display: flex;
    flex-direction: column;
    height: 100%;

    @media screen and (max-width: 400px) {
      clear: both;
      width: 100%;
    }

    @media screen and (max-width: 700px) {
      float: none;
    }
  }
}
