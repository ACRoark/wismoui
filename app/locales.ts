export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
