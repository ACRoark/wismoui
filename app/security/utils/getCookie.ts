const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
