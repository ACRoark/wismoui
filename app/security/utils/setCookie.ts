interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
