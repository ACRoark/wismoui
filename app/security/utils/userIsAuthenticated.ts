import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
