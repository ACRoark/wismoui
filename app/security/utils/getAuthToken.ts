import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
