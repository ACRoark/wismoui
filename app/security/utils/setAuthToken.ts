import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
