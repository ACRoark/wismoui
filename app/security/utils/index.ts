import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
