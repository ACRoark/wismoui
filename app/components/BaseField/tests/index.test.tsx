import React from 'react';
import { MessageDescriptor } from 'react-intl';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import BaseField from '..';

const label: MessageDescriptor = {
  id: 'wismo.components.base-field.tests.label',
  defaultMessage: 'This is just a filler message',
};

const fillerComponent = <div>1234567890</div>;

describe('<BaseField />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<BaseField label={label}>{fillerComponent}</BaseField>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should render correctly when the current language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(<BaseField label={label}>{fillerComponent}</BaseField>, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
