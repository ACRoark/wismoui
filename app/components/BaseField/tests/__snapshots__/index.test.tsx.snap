// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseField /> should render correctly when the current language is en 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Test Label
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div>
      1234567890
    </div>
  </span>
</div>
`;

exports[`<BaseField /> should render correctly when the current language is es 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Etiqueta de prueba
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div>
      1234567890
    </div>
  </span>
</div>
`;
